### DISCUSSION TITLE
Regarding public license

### DISCUSSION THREAD

**Discussion by arvisionlab** [mozilla/hubs#2952] [Hubs Cloud Questions] [answered]
Hi,
i have few query about mozilla public license for Hubs Cloud  (own server host)
- shall i use mozilla hubs cloud for my startup business with my modified version?
- shall i charge for my VR service using hubs ?
- is that required to provide hubs source code with my modified version to public?

**Comment by emclaren** [comment, marked as answer]
Hi @doreai - I believe we corresponded with via hubs@mozilla.com last week and responded to this question. Below you will find the answers we responded with

"Thanks for reaching out. We are not able to provide legal guidance with regards to whether user's applications meet the public licence requirements and recommend that you consult with your own legal council on this front.  You can find the Hubs source code licence here: https://github.com/mozilla/hubs/blob/master/LICENSE and the AWS Hubs Cloud End User agreement here https://d7umqicpi7263.cloudfront.net/eula/product/94ec4ef1-285d-4e63-b2eb-0b65ea93b942/22068f08-c710-4bc7-9f52-4b6a44ad12ef.pdf. I've shared a bit more inline below:

* shall i use mozilla hubs cloud for my startup business with my modified version?
    Yes - you can modify your own instance of Hubs Cloud for your startup business, per the terms and conditions of using Hubs Cloud as stated above

 * shall i charge fee for my VR service using hubs ?
    Yes - you can charge fees for VR services using Hubs. I would refer to the documents above for more information about the terms of service.

  * is that required to provide hubs source code with my modified version to public?
       The Hubs source code license is shared above. We cannot speak specifically to how changes to code may be covered under the MPL and recommend consulting with your own legal council on compliance here. "
